src/backends/odbc/standard-into-type.cpp

```diff
@@ -183,6 +220,32 @@ void odbc_standard_into_type_backend::post_fetch(
                 throw soci_error("Buffer size overflow; maybe got too large string");
             }
         }
+        // Handle the case where the type is a standard wide string (std::wstring)
+        else if (type_ == x_stdwstring)
+        {
+            // Cast the data_ to a reference of type std::wstring
+            std::wstring& s = exchange_type_cast<x_stdwstring>(data_);
+            
+#if defined(SOCI_WCHAR_T_IS_WIDE) // Unices
+            // On Unix-like systems where wchar_t is wide (typically 32-bit)
+            // Convert the UTF-16 buffer to a UTF-32 string
+            std::u32string u32str = utf16_to_utf32(reinterpret_cast<char16_t*>(buf_));
+            // Assign the converted UTF-32 string to the std::wstring
+            s = std::wstring(u32str.begin(), u32str.end());
+#else // Windows
+            // On Windows systems where wchar_t is 16-bit
+            // Directly assign the buffer (interpreted as wchar_t) to the std::wstring
+            s = std::wstring(reinterpret_cast<wchar_t*>(buf_));
+#endif // SOCI_WCHAR_T_IS_WIDE
```

Comment:
I think it would make sense to have a `utf6_to_wide()` function in the Unicode helpers header, this could be useful to the other backends too.